src/util/helperFunctions.ts

```diff
@@ -0,0 +1,44 @@
+export function regexGroupToArray(basis: string, pattern: RegExp, group: number): string[] {
+	const groupContent: string[] = [];
+	let matchGroups: string[] | null;
+
+	while ((matchGroups = pattern.exec(basis)) !== null) {
+		groupContent.push(matchGroups[group]);
+	}
+
+	return groupContent;
+}
+
+export function kahnAlgorithm(graph: Set<string[]>): string[] {
```

Comment:
This exported method is modifying its input. That might be confusing to other users. 
You can clearly communicating that you are not modifying the input, by changing the input type to `ReadonlySet` or `Iterable`, then creating a modifiable set within the function scope.

Reply:
I plan to add JSDocs for later in a different pull-request. I have delayed doing this so far because it didn't show the behavior I expected. I will look into this soon though.